docs/reference/service-pattern/service-pattern.md

```diff
@@ -86,109 +86,98 @@ trait BlobStorage {
 Now, we can implement the `DocRepo` service:
 
 ```scala mdoc:silent
-case class DocRepoImpl(
-    metadataRepo: MetadataRepo,
-    blobStorage: BlobStorage
+final class DocRepoLive(
+  metadataRepo: MetadataRepo,
+  blobStorage: BlobStorage
 ) extends DocRepo {
   override def get(id: String): ZIO[Any, Throwable, Doc] =
-    for {
-      metadata <- metadataRepo.get(id)
-      content <- blobStorage.get(id)
-    } yield Doc(
-      metadata.title,
-      metadata.description,
-      metadata.language,
-      metadata.format,
-      content
-    )
-
+    (metadataRepo.get(id) <&> blobStorage.get(id)).map {
+      case (metadata, content) =>
+        Doc(
+          title = metadata.title,
+          description = metadata.description,
+          language = metadata.language,
+          format = metadata.format,
+          content = content
+        )
+    }
+    
   override def save(document: Doc): ZIO[Any, Throwable, String] =
     for {
-      id <- blobStorage.put(document.content)
-      _ <- metadataRepo.put(
-        id,
-        Metadata(
-          document.title,
-          document.description,
-          document.language,
-          document.format
-        )
+      id       <- blobStorage.put(document.content)
+      metadata = Metadata(
+        title = document.title,
+        description = document.description,
+        language = document.language,
+        format = document.format
       )
+      _        <- metadataRepo.put(id, metadata)
     } yield id
 
-  override def delete(id: String): ZIO[Any, Throwable, Unit] =
-    for {
-      _ <- blobStorage.delete(id)
-      _ <- metadataRepo.delete(id)
-    } yield ()
+  override def delete(id: String): ZIO[Any, Throwable, Unit] = (blobStorage.delete(id) <&> metadataRepo.delete(id)).unit
```

Comment:
I think this is a bit cleaner, wdut?
```suggestion
  override def delete(id: String): ZIO[Any, Throwable, Unit] = blobStorage.delete(id) &> metadataRepo.delete(id).unit
```

Reply:
I'll leave this as is for now since it's super minor and would like to close as many open PRs as possible

Reply:
@kyri-petrou Made the change here: https://github.com/zio/zio/pull/9287